### ISSUE TITLE
Getting DirectoryNotEmpty error when result_bundle option is true.

### ISSUE THREAD

**Issue by rahulsharma2905** [lyndsey-ferguson/fastlane-plugin-test_center#172]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

Issue is occurring in multi_scan action, when result_bundle is true.

'Errno::ENOTEMPTY: [!] Directory not empty' error is shown when second iteration of retry ends.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>bundler: failed to load command: fastlane (PROJECT_DIR/vendor/bundle/ruby/2.3.0/bin/fastlane)
Errno::ENOTEMPTY: [!] Directory not empty @ rb_file_s_rename - (PROJECT_DIR/SCHEME_test_results/HikeXCUITestKitUITests-batch-1/SCHEME.test_result, PROJECT_DIR/SCHEME_test_results/HikeXCUITestKitUITests-batch-1/SCHEME-2.test_result)
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:306:in `rename'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:306:in `move_test_result_bundle_for_next_run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:151:in `handle_test_failure'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:113:in `after_testrun'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:63:in `rescue in run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:51:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:45:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/test_batch_worker.rb:18:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:143:in `block in run_test_batches'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each_with_index'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `run_test_batches'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:53:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-plugin-test_center-3.8.11/lib/fastlane/plugin/test_center/actions/multi_scan.rb:24:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
  Fastfile:16:in `block in parsing_binding'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/gems/fastlane-2.133.0/bin/fastlane:23:in `<top (required)>'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/bin/fastlane:22:in `load'
  PROJECT_DIR/vendor/bundle/ruby/2.3.0/bin/fastlane:22:in `<top (required)>'</pre>
</details>

### Environment

<details>
  <pre>[✔] 🚀 
[16:46:27]: fastlane detected a Gemfile in the current directory
[16:46:27]: however it seems like you don't use `bundle exec`
[16:46:27]: to launch fastlane faster, please use
[16:46:27]: 
[16:46:27]: $ bundle exec fastlane env
[16:46:27]: 
[16:46:27]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[16:46:30]: Generating fastlane environment output, this might take a few seconds...
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.14.6                                                        |
| Ruby                        | 2.3.7                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.20.1 (Apple Git-117)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.14.6 (18G1012)                                     |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.6.5                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 10.2.1                                                         |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
MY_XCODE_WORKSPACE_FILEPATH = File.absolute_path('WORKSPACE')
lane :test_with_retry do |options|
  test_run_block = lambda do |testrun_info|
    failed_test_count = testrun_info[:failed].size
    
    if failed_test_count > 0
      UI.important('The run of tests would finish with failures due to fragile tests here.')

      try_attempt = testrun_info[:try_count]
      if try_attempt < 3
        UI.header('Since we are using :multi_scan, we can re-run just those failing tests!')
      end
    end
  end
  
  result = multi_scan(
    workspace: MY_XCODE_WORKSPACE_FILEPATH,
    try_count: 2,
    fail_build: false,
    scheme: options[:scheme],
    destination: ["#{options[:destination]}"],
    testrun_completed_block: test_run_block,
    result_bundle: true,
    derived_data_path: options[:derived_data_path],
    output_directory: "#{options[:scheme]}_test_results"
  )
  unless result[:failed_testcount].zero?
    UI.message("There are #{result[:failed_testcount]} legitimate failing tests")
  end
end

lane :test do |options|
  scheme_to_test = options[:scheme]
  device_id = options[:device_id]
  scan_options = {
  scheme: scheme_to_test,
  destination: "id=#{device_id}",
  output_directory: "PROJ/#{scheme_to_test}_test_results",
  output_files: 'report.xml',
  output_types: 'junit',
  workspace: 'PROJ/PROJ.xcworkspace',
  result_bundle: true
}
begin
  retry_count ||= 0
  scan(scan_options)
  rescue => e
  UI.important("failure noted: #{e}")
  report_filepath =   File.absolute_path("../PROJ/#{scheme_to_test}_test_results/report.xml")
unless File.exist?(report_filepath)
  raise e
 end
if (retry_count += 1) < 2
    setup_fragile_tests_for_rescan(
     project_path: File.absolute_path('../PROJ/PROJ.xcodeproj'),
     scheme: scheme_to_test,
     report_filepath: report_filepath
     )
    #clear_derived_data
    reset_simulator_contents if retry_count > 1
    retry
   end
  end
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.133.0 | 🚫 Update available |

### Loaded fastlane plugins:

| Plugin                                         | Version | Update-Status      |
| ---------------------------------------------- | ------- | ------------------ |
| fastlane-plugin-setup_fragile_tests_for_rescan | 2.1.1   | ✅ Up-To-Date       |
| fastlane-plugin-test_center                    | 3.8.5   | 🚫 Update available |

<details><summary><b>Loaded gems</b></summary>

| Gem                                            | Version      |
| ---------------------------------------------- | ------------ |
| did_you_mean                                   | 1.0.0        |
| slack-notifier                                 | 2.3.2        |
| atomos                                         | 0.1.3        |
| claide                                         | 1.0.3        |
| colored2                                       | 3.1.2        |
| nanaimo                                        | 0.2.6        |
| xcodeproj                                      | 1.12.0       |
| rouge                                          | 2.0.7        |
| xcpretty                                       | 0.3.0        |
| terminal-notifier                              | 2.0.0        |
| unicode-display_width                          | 1.6.0        |
| terminal-table                                 | 1.8.0        |
| plist                                          | 3.5.0        |
| multipart-post                                 | 2.0.0        |
| word_wrap                                      | 1.0.0        |
| public_suffix                                  | 2.0.5        |
| tty-screen                                     | 0.7.0        |
| tty-cursor                                     | 0.7.0        |
| tty-spinner                                    | 0.9.1        |
| colored                                        | 1.2          |
| highline                                       | 1.7.10       |
| commander-fastlane                             | 4.4.6        |
| faraday                                        | 0.15.4       |
| faraday_middleware                             | 0.13.1       |
| unf_ext                                        | 0.0.7.6      |
| unf                                            | 0.1.4        |
| domain_name                                    | 0.5.20190701 |
| http-cookie                                    | 1.0.3        |
| faraday-cookie_jar                             | 0.0.6        |
| gh_inspector                                   | 1.1.3        |
| mini_magick                                    | 4.9.5        |
| multi_xml                                      | 0.6.0        |
| rubyzip                                        | 1.3.0        |
| security                                       | 0.1.3        |
| xcpretty-travis-formatter                      | 1.0.0        |
| dotenv                                         | 2.7.5        |
| bundler                                        | 2.0.2        |
| jwt                                            | 2.1.0        |
| uber                                           | 0.1.0        |
| declarative                                    | 0.0.10       |
| declarative-option                             | 0.1.0        |
| representable                                  | 3.0.4        |
| retriable                                      | 3.1.2        |
| multi_json                                     | 1.13.1       |
| signet                                         | 0.11.0       |
| memoist                                        | 0.16.0       |
| os                                             | 1.0.1        |
| googleauth                                     | 0.6.7        |
| httpclient                                     | 2.8.3        |
| google-api-client                              | 0.23.9       |
| digest-crc                                     | 0.4.1        |
| google-cloud-storage                           | 1.16.0       |
| emoji_regex                                    | 1.0.1        |
| json                                           | 2.2.0        |
| io-console                                     | 0.4.5        |
| excon                                          | 0.67.0       |
| CFPropertyList                                 | 3.0.1        |
| libxml-ruby                                    | 2.9.0        |
| babosa                                         | 1.0.3        |
| fastimage                                      | 2.1.7        |
| addressable                                    | 2.7.0        |
| psych                                          | 2.1.0.1      |
| mime-types                                     | 3.3          |
| mime-types-data                                | 3.2019.0904  |
| fastlane-plugin-setup_fragile_tests_for_rescan | 2.1.1        |
| xctest_list                                    | 1.1.8        |
| colorize                                       | 0.8.1        |
| fastlane-plugin-test_center                    | 3.8.5        |
</details>

*generated on:* **2019-11-18**
</details>
[16:46:34]: Take notice that this output may contain sensitive information, or simply information that you don't want to make public.
[16:46:34]: Open https://github.com/fastlane/fastlane/issues/new to submit a new issue ✅

#######################################################################
# fastlane 2.136.0 is available. You are on 2.133.0.
# You should use the latest version.
# Please update using `sudo gem install fastlane`.
#######################################################################</pre>
</details>

**Comment by rahulsharma2905** [comment]
I am using Xcode 10.3, i think recent changes for Xcode 11 in file lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb are causing this issue.

**Comment by lyndsey-ferguson** [comment]
@rahulsharma2905 you are not running `multi_scan` in parallel are you? This is similar to #169.

**Comment by rahulsharma2905** [comment]
Not running in parallel, just a single instance.

**Comment by lyndsey-ferguson** [comment]
They still may be the same problem, but I will wait a little while longer to get some feedback from the other person.

**Comment by lyndsey-ferguson** [comment]
It works fine for me, would you be able to provide a sample project and sample `Fastfile` that I can use to debug the issue?

**Comment by rahulsharma2905** [comment]
which version of Xcode are you using? looking at the code i have a feeling that it would work fine for XCode 11.x but not for lower versions.

**Comment by rahulsharma2905** [comment]
Few more findings:
I inserted some logging in move_test_result_bundle_for_next_run method in retrying_scan_helper.rb file. My changes looks like below:
```
def move_test_result_bundle_for_next_run
          return unless @options[:result_bundle]

          glob_pattern = "#{output_directory}/*.test_result"
          preexisting_test_result_bundles = Dir.glob(glob_pattern)
          puts("Test Run Count : #{@testrun_count}")
          puts("Preexisting folders : #{preexisting_test_result_bundles}")
          unnumbered_test_result_bundles = preexisting_test_result_bundles.reject do |test_result|
            test_result =~ /.*-\d+\.test_result/
          end
          src_test_bundle = unnumbered_test_result_bundles.first
          dst_test_bundle_parent_dir = File.dirname(src_test_bundle)
          dst_test_bundle_basename = File.basename(src_test_bundle, '.test_result')
          dst_test_bundle = "#{dst_test_bundle_parent_dir}/#{dst_test_bundle_basename}-#{@testrun_count}.test_result"
          FastlaneCore::UI.verbose("Moving test_result '#{src_test_bundle}' to '#{dst_test_bundle}'")
          File.rename(src_test_bundle, dst_test_bundle)
          preexisting_test_result_bundles = Dir.glob(glob_pattern)
          puts("Preexisting folders after : #{preexisting_test_result_bundles}")
        end
```
In logs i found below lines:
for first iteration:
```
Test Run Count : 1
Preexisting folders : ["PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME.test_result"]
Preexisting folders after : ["PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-1.test_result"]
```
for second iteration:
```
Test Run Count : 2
Preexisting folders : ["PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME.test_result", "PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-2.test_result"]
```
command failed before reaching 'Preexisting folders after' line of code.
Error thrown:
```
Directory not empty @ rb_file_s_rename - (PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME.test_result, PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-2.test_result)
```

**Comment by rahulsharma2905** [comment]
seems like in between calls of this method for iteration 1 and iteration 2, somehow folder PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-1.test_result is renamed to folder PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-2.test_result

**Comment by lyndsey-ferguson** [comment]
Can you re-run your lane with the `--verbose` flag and include the output? [This](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/lib/fastlane/plugin/test_center/helper/multi_scan_manager/report_collator.rb#L133) may be contributing to the problem, although it still works fine for me.

**Comment by rahulsharma2905** [comment]
i am getting this piece of information in logs after running with --verbose:
```
Preexisting folders : ["PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME.test_result"]
DEBUG [2019-11-19 19:48:13.46]: Moving test_result 'PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME.test_result' to 'PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-1.test_result'
Preexisting folders after : ["PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-1.test_result"]
DEBUG [2019-11-19 19:48:13.46]: ReportCollator collating
DEBUG [2019-11-19 19:48:13.47]: Copying test_result bundle PROJECT_DIR/SCHEME_test_results/PROJECT-batch-1/SCHEME-1.test_result
DEBUG [2019-11-19 19:48:13.47]: Deleting xcresults:
DEBUG [2019-11-19 19:48:13.47]: Restarting Simulator 89E34003
INFO [2019-11-19 19:48:14.23]: Starting scan #2 with 1 tests for batch #1.
```

**Comment by lyndsey-ferguson** [comment]
@rahulsharma2905 I have duplicated the error, now I can look into it more.

**Comment by lyndsey-ferguson** [comment]
I think it could be the same as the other Issue as I can reproduce it in single mode. I will close this Issue and link the two. The other issue is [#169](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/169)
